### PR TITLE
Add clipseg models

### PR DESCRIPTION
**Pull request by w5688414**
### PR types

+ New features

### PR changes

+ Models

### Description

```
from paddlenlp.transformers import CLIPSegProcessor
from paddlenlp.transformers import CLIPSegTextConfig, CLIPSegConfig
from paddlenlp.transformers import CLIPSegForImageSegmentation
import paddle.nn.functional as F
from PIL import Image
import requests
url = "http://images.cocodataset.org/val2017/000000039769.jpg"
model_name_or_path ='CIDAS/clipseg-rd64-refined'
config = CLIPSegConfig.from_pretrained(model_name_or_path)
processor = CLIPSegProcessor.from_pretrained(model_name_or_path)
model = CLIPSegForImageSegmentation.from_pretrained(model_name_or_path)
model.eval()
image = Image.open("000000039769.jpg")
texts = ["a cat", "a remote", "a blanket"]
inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pd")
with paddle.no_grad():
    outputs = model(**inputs, return_dict=True)
logits = outputs.logits
print(logits.shape)

import matplotlib.pyplot as plt

logits = logits.unsqueeze (1)
_, ax = plt.subplots (1, len (texts) + 1, figsize=(3*(len (texts) + 1), 12))
[a.axis ('off') for a in ax.flatten ()]
ax [0].imshow (image)
[ax [i+1].imshow (F.sigmoid(logits [i][0]).numpy()) for i in range (len (texts))]
[ax [i+1].text (0, -15, prompt) for i, prompt in enumerate (texts)]
plt.savefig('output_pd.png')
```
Paddle:
output_pd
torch:

output_pt

### CODE REVIEW

File: paddlenlp/transformers/clipseg/processing.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            visual_prompt (`PIL.Image.Image`, `np.ndarray`, `torch.Tensor`, `List[PIL.Image.Image]`, `List[np.ndarray]`, `List[torch.Tensor]`):
+                The visual prompt image or batch of images to be prepared. Each visual prompt image can be a PIL image,
+                NumPy array or PyTorch tensor. In case of a NumPy array/PyTorch tensor, each image should be of shape
+                (C, H, W), where C is a number of channels, H and W are image height and width.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*):
+                If set, will return tensors of a particular framework. Acceptable values are:
+                - `'tf'`: Return TensorFlow `tf.constant` objects.
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return NumPy `np.ndarray` objects.
+                - `'jax'`: Return JAX `jnp.ndarray` objects.
```
Comment (guoshengCS) [inline paddlenlp/transformers/clipseg/processing.py]: 注意调整API文档

File: paddlenlp/transformers/clipseg/modeling.py [hunk 2]
```diff
@@ -0,0 +1,1355 @@
...
+
+# Copied from transformers.models.bart.modeling_bart._expand_mask
+def _expand_mask(mask: paddle.Tensor, tgt_len: Optional[int] = None):
+    """
+    Expands attention_mask from `[bsz, seq_len]` to `[bsz, 1, tgt_seq_len, src_seq_len]`.
+    """
+    bsz, src_len = mask.shape
+    tgt_len = tgt_len if tgt_len is not None else src_len
+
+    expanded_mask = mask[:, None, None, :].expand([bsz, 1, tgt_len, src_len])
```
Comment (guoshengCS) [inline paddlenlp/transformers/clipseg/modeling.py:57]: paddle的expand和torch的有区别

File: paddlenlp/transformers/clipseg/modeling.py [hunk 3]
```diff
@@ -0,0 +1,1355 @@
...
+        ```python
+        >>> from paddlenlp.transformers import AutoProcessor, CLIPSegForImageSegmentation
+        >>> from PIL import Image
+        >>> import requests
+        >>> processor = AutoProcessor.from_pretrained("CIDAS/clipseg-rd64-refined")
+        >>> model = CLIPSegForImageSegmentation.from_pretrained("CIDAS/clipseg-rd64-refined")
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> texts = ["a cat", "a remote", "a blanket"]
+        >>> inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pt")
```
Comment (guoshengCS) [inline paddlenlp/transformers/clipseg/modeling.py]: 注意清理API文档

File: paddlenlp/transformers/clipseg/processing.py (hunk 1 above)
Comment (w5688414) [reply paddlenlp/transformers/clipseg/processing.py]: 已修改

File: paddlenlp/transformers/clipseg/modeling.py (hunk 3 above)
Comment (w5688414) [reply paddlenlp/transformers/clipseg/modeling.py]: 已修改

**Review by guoshengCS — DISMISSED** [review]

**Review by guoshengCS — APPROVED** [review]
